quant: clean up normalization channels_last handling

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39413 graph mode: util for fusion of functions which require observation
* #39203 quant layer/group/instance norm: make weights and biases optional
* #39096 instancenorm: static quant graph mode support
* #39095 groupnorm: graph mode static quant support
* #39094 layernorm: eager mode qat support
* #39093 instancenorm: eager mode QAT support
* #39092 groupnorm: eager mode QAT support
* #39091 instancenorm: eager static quant support
* #39090 groupnorm: eager static quant support
* **#37802 quant: clean up normalization channels_last handling**

Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

vkuzo committed:
quant: clean up normalization channels_last handling

Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

jerryzh168 commented on aten/src/ATen/native/quantized/cpu/qnormalization.cpp:
```diff
@@ -55,13 +55,17 @@ Tensor quantized_group_norm_impl(
     int64_t output_zero_point) {
 
-  const auto input_ndim = qx.dim();
+  const auto& qx_contig = qx.is_contiguous() ? qx : qx.contiguous();
+  const auto& weight_contig = weight.is_contiguous() ? weight : weight.contiguous();
+  const auto& bias_contig = bias.is_contiguous() ? bias : bias.contiguous();
```
this is probably the same as just always calling contiguous: https://codebrowser.bddppq.com/pytorch/pytorch/aten/src/ATen/native/TensorProperties.cpp.html#83

jerryzh168 replied:
this comment is still valid

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
LG

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]

vkuzo committed:
Update on "quant: clean up normalization channels_last handling"


Summary:

* adds test coverage for channels_last input format for quantized normalization ops
* fixes quantized group_norm and instance_norm to always return contiguous tensors

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_group_norm
python test/test_quantization.py TestQuantizedOps.test_qlayer_norm
python test/test_quantization.py TestQuantizedOps.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21395196](https://our.internmc.facebook.com/intern/diff/D21395196)

[ghstack-poisoned]
